src/qt/receivecoinsdialog.cpp

```diff
@@ -144,9 +141,14 @@ void ReceiveCoinsDialog::on_receiveButton_clicked()
     QString address;
     QString label = ui->reqLabel->text();
     /* Generate new receiving address */
-    OutputType address_type = model->getDefaultAddressType();
-    if (address_type != OUTPUT_TYPE_LEGACY) {
-        address_type = ui->useBech32->isChecked() ? OUTPUT_TYPE_BECH32 : OUTPUT_TYPE_P2SH_SEGWIT;
+    OutputType address_type;
+    if (ui->useBech32->isChecked()) {
+        address_type = OUTPUT_TYPE_BECH32;
+    } else {
+        address_type = model->getDefaultAddressType();
+        if (address_type == OUTPUT_TYPE_BECH32) {
+            address_type = OUTPUT_TYPE_P2SH_SEGWIT;
```

Comment:
I don't understand the logic here: if default address type is BECHS32 set it to P2SH_SEGWIT

Reply:
because `ui->useBech32->isChecked()` is false in line 146, no?

Reply:
Indeed. The logic is that if useBech32 is unchecked, we don't want to use Bech32 even if that's the user's default. In that case, we use P2SH_SEGWIT instead. But if the default is P2SH_SEGWIT or LEGACY, then we leave it alone.

| Addresstype default | Checkbox | Result |
|---|----|----|
| legacy | on | Bech32 |
| p2sh-segwit | on | Bech32 |
| bech32 | on | Bech32 |
| legacy | off | Legacy |
| p2sh-segwit | off | P2SH-Segwit |
| bech32 | off | P2SH-Segwit |

Reply:
That makes sense.